### ISSUE TITLE
Gatsby CLI to add scaffolding for source plugin creation

### ISSUE THREAD

**Issue by shannonbux** [gatsbyjs/gatsby#13376]
## Summary

This is an idea to add to command to Gatsby CLI for people starting to build a source plugin.

It could be `gatsby start source-plugin` or `gatsby new plugin` or `gatsby plugin --help` or something.

### Basic example

```
Watsons-MacBook-Pro:~ watsondog$ gatsby start source-plugin

Welcome to the Gatsby source plugin help center!

Resources:
- Walk through the tutorial (https://www.stuff.com)
- Read requirements for Gatsby packages (file and naming conventions)  (https://www.stuff.com)
- Submit to plugin library (https://www.stuff.com)
- Read best practices (https://www.stuff.com)
- Join Discord #plugin-authoring channel to ask questions (https://www.stuff.com)

Run `gatsby start source-plugin` anytime you'd like to see this help center again. Thank you!
```

This is the most basic version of how we could help plugin authors. We could do more, such as walk them step by step into creating a local plugins folder, etc. or create it for them, etc.

### Motivation

People say the plugin authoring docs are useful but hardly anyone looks at them. This could increase views of docs and decrease the amount of time it takes to create a plugin. The trick then would be for people to find out about the new command `gatsby start source-plugin`. #12951 

Thx and fyi @DSchau @jlengstorf @marcysutton @KyleAMathews @prestonso @calcsam

**Comment by frinyvonnick** [comment]
I'll start working on this if it is good to go ? Can I assign myself to it ?

**Comment by frinyvonnick** [comment]
Closed by #13450 😄
